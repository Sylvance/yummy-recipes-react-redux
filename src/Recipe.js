import React, { Component } from 'react';
import logo from './logo.svg';
import './Recipe.css';

class Recipe extends Component {
  render() {
    return (
      <div className="Recipe">
        <header className="Recipe-header">
          <img src={logo} className="Recipe-logo" alt="logo" />
          <h1 className="Recipe-title">Recipes Page</h1>
        </header>
        <p className="Recipe-intro">
          To get started, edit <code>src/Recipe.js</code> and save to reload.
        </p>
        <div className="Recipe-intro">
          Yummy recipes provides a platform for users to keep track of their awesome 
          recipes and share with others if they so wish.
        </div>
      </div>
    );
  }
}

export default Recipe;
