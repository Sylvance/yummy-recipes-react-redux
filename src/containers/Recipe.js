import React from 'react'
import PropTypes from 'prop-types'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'

import RecipeHeader from '../components/RecipeHeader'
import RecipeMainSection from '../components/RecipeMainSection'

import * as Actions from '../actions'

const Recipe = ({recipes, categoryid, meta, actions, props}) => (
  <div>
    <RecipeHeader addRecipe={actions.addRecipe} categoryid={categoryid} />
    <RecipeMainSection categoryid={categoryid} meta={meta} recipes={recipes} actions={actions} />
  </div>
)

Recipe.propTypes = {
  recipes: PropTypes.array.isRequired,
  categoryid: PropTypes.number.isRequired,
  actions: PropTypes.object.isRequired,
  meta: PropTypes.object.isRequired
}

const mapStateToProps = (state, props) => ({
  recipes: state.recipes,
  meta: state.meta,
  categoryid: parseInt(props.match.params.categoryid, 10)
})

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(Actions, dispatch)
})

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Recipe)
