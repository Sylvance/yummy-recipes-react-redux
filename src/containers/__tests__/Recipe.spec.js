// Auto-generated do not edit

/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import Recipe from '../Recipe';


describe('Recipe test', () => {
  it('Recipe should match snapshot', () => {
    const component = renderer.create(
        <div>
      </div>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
