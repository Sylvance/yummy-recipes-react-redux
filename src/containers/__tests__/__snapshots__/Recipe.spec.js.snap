// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe test Recipe should match snapshot 1`] = `<div />`;
